src/device-state.ts

```diff
@@ -344,6 +350,11 @@ export async function setTarget(target: TargetState, localSource?: boolean) {
 			await config.set({ targetStateSet: true }, trx);
 		});
 	});
+
+	await fsUtils.writeAndSyncFile(
```

Comment:
Sorry, one last request. Could you move this just above the `usingWriteLockTarget` call? That call can take a while and it's better that the configuration is dumped earlier in the process.

Reply:
Done